src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double DEFAULT_MULTIPLICATION_NUMBER = 1.5;
+    private static final int START_ARRAY_SIZE = 0;
+    private T[] elementsData;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        this.elementsData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.capacity = DEFAULT_CAPACITY;
+        this.size = START_ARRAY_SIZE;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() == capacity) {
+            resizeArray();
+        }
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size() == capacity) {
+            resizeArray();
+        }
+        if (index >= 0 && index <= size) {
```

Comment:
```suggestion
        checkIndexForAdd(...
        System.arraycopy(...
```

Reply:
Create method `checkIndexForAdd(index)`
use this method
implement other logic 


Don't use if else here in such way